src/main/java/bsh/BSHArrayInitializer.java

```diff
@@ -145,6 +147,7 @@ private Object buildArray(int dimensions, Class<?> baseType,
         // the values are set.
         dims[0] = jjtGetNumChildren();
         Object array =  Array.newInstance( baseType, dims );
+        Class<?> entryType = array.getClass().getComponentType();
```

Comment:
Isn't this what `arrayElementType` is supposed to do? I am still curious if that is not broken.

Reply:
Do you mean Types.arrayELementType()?  That will return the ultimate base type of the array, so for int[][] it will return int.  However here we want the type for the current dimension, which would be int[].